metricsaggregator/src/main/java/io/dockstore/metricsaggregator/MetricsAggregatorAthenaClient.java

```diff
@@ -0,0 +1,342 @@
+        LOG.info("Creating table: {}", tableName);
+        final String entityProjectionValues = String.join(",", metadataApi.getEntryTypeMetadataList()
+                .stream()
+                .map(EntryTypeMetadata::getTerm)
+                .toList());
+        final String registryProjectionValues = Stream.concat(
+                    metadataApi.getSourceControlList().stream().map(SourceControlBean::getValue),
+                    metadataApi.getDockerRegistries().stream().map(RegistryBean::getDockerPath))
+                .collect(Collectors.joining(","));
+        final String platformProjectionValues = String.join(",", Arrays.stream(Partner.values()).map(Partner::name).toList());
+        final String query = String.format("""
+                CREATE EXTERNAL TABLE IF NOT EXISTS %s (
+                    runexecutions array<struct<
+                        executionid:string,
+                        dateexecuted:string,
+                        executionstatus:string,
+                        executiontime:string,
+                        memoryrequirementsgb:double,
+                        cpurequirements:int,
+                        cost:struct<value:double,currency:string>,
+                        region:string,
+                        additionalproperties:string
+                        >
+                    >,
+                    taskexecutions array<struct<
+                        executionid:string,
+                        taskexecutions:array<struct<
+                            executionid:string,
+                            dateexecuted:string,
+                            executionstatus:string,
+                            executiontime:string,
+                            memoryrequirements:double,
+                            cpurequirements:int,
+                            cost:struct<value:double,currency:string>,
+                            region:string,
+                            additionalproperties:string
+                            >
+                        >
+                    >>,
+                    validationexecutions array<struct<
+                        executionid:string,
+                        dateexecuted:string,
+                        validatortool:string,
+                        validatortoolversion:string,
+                        isvalid:boolean,
+                        errormessage:string,
+                        additionalproperties:string
+                        >
+                    >
+                )
+                PARTITIONED BY (
+                    `entity` string,
+                    `registry` string,
+                    `org` string,
+                    `name` string,
+                    `version` string,
+                    `platform` string
+                )
+                ROW FORMAT SERDE 'org.openx.data.jsonserde.JsonSerDe'
+                LOCATION 's3://%s/'
+                TBLPROPERTIES (
+                    "projection.enabled" = "true",
```

Comment:
This uses [partition projection](https://docs.aws.amazon.com/athena/latest/ug/partition-projection.html) to automatically create partitions. We don't have to add partitions, this definition takes care of it all